### PR TITLE
NOAA GSL updates to output new simulated satellite brightness temps

### LINKED ISSUE #571: NOAA GSL UPP updates
NOAA GSL has a few UPP updates from our RRFS_dev branch.  The development has occurred since early 2022.  We plan to break the updates into several smaller PRs.  Changes include:

(1) Small changes to GRIB2 encoding for some variables that were causing a wgrib2 error

(2) Implementing RAP/HRRR - like smoothing in RRFS

(3) Small modification to pass through soil moisture availability for RRFS

(4) Adding a check for extremely large positive or negative UH values in CALUPDHEL.f

(5) Introducing new fields showing QPF exceeding average recurrence intervals

(6) Adding GOES-18 simulated satellite brightness temps (from Will Lewis - SSEC)

(7) A few small RTMA-related bug fixes from Edward Colon

### PR DESCRIPTION
**Pull request by EricJames-NOAA**
This PR includes updates to UPP to output new simulated satellite brightness temperatures from the RRFS.  The code was developed by Will Lewis from Wisconsin / SSEC, and merged into GSL RRFS_dev branch during 2022.  It is now used in realtime RRFS_NA_3km system at GSL.   Tests on Jet show identical output for RRFS except for the new fields.

Ka Yee Wong ran UPP regression tests on Hera, and the only non bitwise identical results are in the naming of the soil moisture content variable:

[Eric.James@hfe09 ejames]$ diff sfluxgrbf006_pe.txt sfluxgrbf006_pe_hera.txt
161c161
< 21:47496981:vt=2021021606:0-2 m below ground:6 hour fcst:SOILM Soil Moisture Content [kg/m^2]:
>
21:47496981:vt=2021021606:0-2 m below ground:6 hour fcst:SOIL_M Soil Moisture [kg/m^3]:

This discrepancy shows up for slightly more than half of the regression tests.

### CODE REVIEW

**Comment by WenMeng-NOAA** [comment]
@EricJames-NOAA Can you open a UPP issue to link this PR? Thanks!

File: parm/params_grib2_tbl_new [hunk 1]
```diff
@@ -812,6 +812,16 @@
    3 192    36   0 SBTA177
    3 192    37   0 SBTA178
    3 192    38   0 SBTA179
+   3 192    79   0 SBTA1810
```
Comment (WenMeng-NOAA) [inline parm/params_grib2_tbl_new:815]: @EricJames-NOAA Can you add these new grib2 names in parm/params_grib2_tbl_new.text which is the un-sorted grib2 table. You may refer to the table at https://github.com/NOAA-EMC/NCEPLIBS-g2tmpl/blob/e1397436a1584180fca755103e65ebce98d14619/src/params_grib2_tbl_new.text#L1318
We would like to keep these sorted and un-sorted tables consistent.

File: sorc/ncep_post.fd/CALRAD_WCLOUD_newcrtm.f [hunk 2]
```diff
@@ -2078,6 +2116,30 @@ SUBROUTINE CALRAD_WCLOUD
                    endif
                  enddo ! channel loop
               end if  ! end of outputting goes 17
+!             Wm Lewis updated idx for g18 on 3 JUN 2022
+              if (isis=='abi_g18')then  ! writing goes 16 to grib
```
Comment (WenMeng-NOAA) [inline sorc/ncep_post.fd/CALRAD_WCLOUD_newcrtm.f]: @EricJames-NOAA I would assume the comments for "writing goes 18"?

Comment (EricJames-NOAA) [reply sorc/ncep_post.fd/CALRAD_WCLOUD_newcrtm.f]: @WenMeng-NOAA good catch!  I just corrected that, and an earlier incorrect comment for writing goes 17.

**Comment by WenMeng-NOAA** [comment]
@EricJames-NOAA I ran the UPP standalone test with model data files you provided before and UPP control file from  your PR. It appears to me simulated satellite products were generated. My test results are at /scratch1/NCEPDEV/stmp2/Wen.Meng/rrfs_2021100815
Can you take a look at and let me know if they are inconsistent with your test results.

**Comment by EricJames-NOAA** [comment]
@WenMeng-NOAA the results look good to me!  Thank you for your test.

**Review by WenMeng-NOAA — APPROVED** [review]

**Comment by WenMeng-NOAA** [comment]
The UPP RT tests were completed on WCOSS2, Hera and Orion. No baseline update is needed from this PR.

**Pull request merged**